src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/NativeTracerContext.cs

```diff
@@ -0,0 +1,12 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native;
+
+public struct NativeTracerContext(Address? beneficiary)
+{
+    public Address From;
+    public Address? To;
+    public readonly Address? Beneficiary = beneficiary;
```

Comment:
Can't those be inferred from `TxExecutionContext` and `BlockExecutionContext` accesible from `ExecutionEnvironment`? If not maybe add them there?

Reply:
Yep, they can be inferred through `TxExecutionContext` if I add them as properties in that struct and add them while building the execution environment. I'll then need to add the execution environment as a param for the `ReportAction` method so that I can pull off the transaction `from` address, `to` address, and `beneficiary` address on the first call. I think it would be worthwhile exposing the execution environment for ReportAction as well in case we want to add any new native tracers in the future that could take advantage of that so I'll go ahead and make this change.

Reply:
I added the spike for this approach on a [commit](https://github.com/NethermindEth/nethermind/commit/07ccd6dd60dbd8c1a87de4d4544d7b8a46a4e48d) in a separate branch since I realized that having this logic in the `ReportAction` method would be kind of hacky since the nonce for the transaction sender is incremented prior to entering ReportAction and would need special handling for the initial case to decrement it by one. I thought about exposing another trace method to the ITxTracer API to mirror the geth `OnTxStart` hook, but felt like that might be overkill for this use case since the information can just be passed from the constructor.